client/ayon_core/pipeline/stagingdir.py

```diff
@@ -0,0 +1,220 @@
+    if STAGING_DIR_TEMPLATES not in anatomy.templates:
+        raise ValueError(
+            (
+                'Anatomy of project "{}" does not have set' ' "{}" template section!'
+            ).format(project_name, template_name)
+        )
+
+    if template_name not in anatomy.templates[STAGING_DIR_TEMPLATES]:
+        raise ValueError(
+            (
+                'Anatomy of project "{}" does not have set'
+                ' "{}" template key at Staging Dir section!'
+            ).format(project_name, template_name)
+        )
+
+
+def get_staging_dir(
+    host_name,
+    project_entity,
+    folder_entity,
+    task_entity,
+    product_type,
+    product_name,
+    anatomy,
+    project_settings=None,
+    **kwargs
+):
+    """Get staging dir data.
+
+    If `force_temp` is set, staging dir will be created as tempdir.
+    If `always_get_some_dir` is set, staging dir will be created as tempdir if
+    no staging dir profile is found.
+    If `prefix` or `suffix` is not set, default values will be used.
+
+    Arguments:
+        host_name (str): Name of host.
+        project_entity (Dict[str, Any]): Project entity.
+        folder_entity (Dict[str, Any]): Folder entity.
+        task_entity (Dict[str, Any]): Task entity.
+        product_type (str): Type of product.
+        product_name (str): Name of product.
+        anatomy (ayon_core.pipeline.Anatomy): Anatomy object.
+        project_settings (Optional[Dict[str, Any]]): Prepared project settings.
+        **kwargs: Arbitrary keyword arguments. See below.
+
+    Keyword Arguments:
+        force_temp (bool): If True, staging dir will be created as tempdir.
+        always_return_path (bool): If True, staging dir will be created as
+            tempdir if no staging dir profile is found.
+        prefix (str): Prefix for staging dir.
+        suffix (str): Suffix for staging dir.
+        formatting_data (Dict[str, Any]): Data for formatting staging dir
+            template.
+
+    Returns:
+        Dict[str, Any]: Staging dir data
+    """
+
+    log = kwargs.get("log") or Logger.get_logger("get_staging_dir")
+    always_return_path = kwargs.get("always_return_path")
+
+    # make sure always_return_path is set to true by default
+    if always_return_path is None:
+        always_return_path = True
+
+    if kwargs.get("force_temp"):
+        return get_temp_dir(
+            project_name=project_entity["name"],
+            anatomy=anatomy,
+            prefix=kwargs.get("prefix"),
+            suffix=kwargs.get("suffix"),
+        )
+
+    # making fewer queries to database
+    ctx_data = get_template_data(
+        project_entity, folder_entity, task_entity, host_name
+    )
+    # add roots to ctx_data
+    ctx_data["root"] = anatomy.roots
+
+    # add additional data
+    ctx_data.update({
+        "product": {
+            "type": product_type,
+            "name": product_name
+        },
+        "host": host_name,
```

Comment:
This is adding new key to available keys. This should be maybe added in `get_template_data` instead and removed from here?